### ISSUE TITLE
[1.5.0-beta.3] Bump to ng 5 breaks compilation (ts.createCallSignature is not a function)

### ISSUE THREAD

**Issue by cexbrayat** [angular/angular-cli#7924]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
@angular/cli: 1.5.0-beta.3
node: 8.6.0
os: darwin x64
@angular/animations: 5.0.0-rc.0
@angular/common: 5.0.0-rc.0
@angular/compiler: 5.0.0-rc.0
@angular/core: 5.0.0-rc.0
@angular/forms: 5.0.0-rc.0
@angular/http: 5.0.0-rc.0
@angular/platform-browser: 5.0.0-rc.0
@angular/platform-browser-dynamic: 5.0.0-rc.0
@angular/router: 5.0.0-rc.0
@angular/cli: 1.5.0-beta.3
@angular/compiler-cli: 5.0.0-rc.0
@angular/language-service: 5.0.0-rc.0
typescript: 2.5.3

### Repro steps.
New app generated with 1.5.0-beta.3
Bump Angular packages to 5.0.0-rc.0

### The log given by the failure.
```
ERROR in TypeError: ts.createCallSignature is not a function
    at Object.replaceResources (/Users/cedric/Code/ninjasquad/formation/Angular2/ponyracer/node_modules/@angular/cli/node_modules/@ngtools/webpack/src/transformers/replace_resources.js:45:16)
    at sourceFiles.forEach (/Users/cedric/Code/ninjasquad/formation/Angular2/ponyracer/node_modules/@angular/cli/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:515:61)
    at Array.forEach (<anonymous>)
    at Promise.resolve.then.then.then (/Users/cedric/Code/ninjasquad/formation/Angular2/ponyracer/node_modules/@angular/cli/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:511:29)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```
I tried with different TS versions, but still have the error

**Comment by vecernik** [comment]
same with 1.5.0-beta.4 and angular 5.0.0-rc.1

**Comment by Destreyf** [comment]
I was running into the same issue under Debian, i swapped typescript to 2.4.2 and it seems to build properly, simply running `npm i typescript@2.4.2` then running `ng serve` worked.

**Comment by cexbrayat** [comment]
To be more accurate, I use Yarn as dependency manager: installing TS 2.3, 2.4 or 2.5 doesn't change the result as the CLI relies on 2.3.4

```
❯ yarn list typescript                                                                                                                              
yarn list v1.1.0
├─ @angular-devkit/build-optimizer@0.0.25
│  └─ typescript@2.3.4
├─ @angular/cli@1.5.0-beta.3
│  └─ typescript@2.3.4
└─ typescript@2.4.2
```

**Comment by cexbrayat** [comment]
I think the issue is fixed with 1.5.0-rc.0 and 5.0.0-rc.2 (at least I can't reproduce it anymore).

**Comment by filipesilva** [comment]
This issue can come up if the package manager decides to not dedupe `typescript`.

Angular CLI depends on `"typescript": ">=2.0.0 <2.6.0"`, while the project also has a typescript dependency. If the package manager decides that it doesn't really need to dedupe typescript, a project can end up with an up to date version of TS for the project (like `2.5.2`) but an older TS for the CLI itself (like `2.3.4`). 

Then when `@ngtools/webpack` tries to get typescript, it gets the 2.3.4 and fails because it does not contain `ts.createCallSignature`.

This situation is unfortunately not easily fixable without a breaking change. To do these dependencies properly, we'd need to split the global and local CLI in different packages, with different peer dependencies.

The reason why the problem can go away on it's own is because successive reinstalls can force the package manager dedupe typescript.
